src/vs/platform/terminal/common/capabilities/capabilities.ts

```diff
@@ -216,24 +216,40 @@ export interface IPartialCommandDetectionCapability {
 	readonly onCommandFinished: Event<IXtermMarker>;
 }
 
-export interface ITerminalCommand {
+interface IBaseTerminalCommand {
+	// Mandatory
 	command: string;
 	isTrusted: boolean;
 	timestamp: number;
-	cwd?: string;
-	exitCode?: number;
+
+	// Optional serializable
```

Comment:
did you go with `| undefined` instead of `property?` for a reason here?

Reply:
`| undefined` is stricter and I wanted to use it on the serializable one, it worked for `ITerminalCommand` too.